app/utils/users.ts

```diff
@@ -14,10 +14,7 @@ const INITIAL_PAGE_LIMIT = 10;
 const fetcher = async (url: string, token: string) =>
   fetch(url, {
     headers: { Authorization: `Bearer ${token}` },
-  }).then((res) => {
-    if (!res.ok) throw new Error("Failed to fetch");
-    return res.json();
-  });
+  }).then((res) => res.json());
```

Comment:
this reworking of error message is to better surface the errors from our API rather than throwing hte generic "Failed to fetch"